app/api/items/update_item.feature

```diff
@@ -0,0 +1,186 @@
+
+Background:
+  Given the database has the following table 'users':
+    | ID | sLogin | tempUser | idGroupSelf | idGroupOwned |
+    | 1  | jdoe   | 0        | 11          | 12           |
+  And the database has the following table 'groups':
+    | ID | sName      | sType     |
+    | 11 | jdoe       | UserSelf  |
+    | 12 | jdoe-admin | UserAdmin |
+  And the database has the following table 'items':
+    | ID |
+    | 21 |
+    | 50 |
+    | 60 |
+  And the database has the following table 'items_items':
+    | idItemParent | idItemChild |
+    | 21           | 60          |
+    | 50           | 21          |
+  And the database has the following table 'items_ancestors':
+    | idItemAncestor | idItemChild |
+    | 21             | 60          |
+    | 50             | 21          |
+    | 50             | 60          |
+  And the database has the following table 'groups_items':
+    | ID | idGroup | idItem | bManagerAccess | bOwnerAccess |
+    | 40 | 11      | 50     | false          | true         |
+    | 41 | 11      | 21     | true           | false        |
+    | 42 | 11      | 60     | false          | true         |
+  And the database has the following table 'groups_ancestors':
+    | ID | idGroupAncestor | idGroupChild | bIsSelf |
+    | 71 | 11              | 11           | 1       |
+    | 72 | 12              | 12           | 1       |
+  And the database has the following table 'languages':
+    | ID |
+    | 3  |
+
+Scenario: Valid
+  Given I am the user with ID "1"
+  And the time now is "2018-01-01T00:00:00Z"
+  When I send a PUT request to "/items/50" with the following body:
+    """
+    {
+      "type": "Course"
+    }
+    """
+  Then the response should be "updated"
+  And the table "items" at ID "50" should be:
+    | ID | sType  | sUrl | idDefaultLanguage | bTeamsEditable | bNoScore | sTextID | bTitleBarVisible | bCustomChapter | bDisplayDetailsInParent | bUsesAPI | bReadOnly | sFullScreen | bShowDifficulty | bShowSource | bHintsAllowed | bFixedRanks | sValidationType | iValidationMin | idItemUnlocked | iScoreMinUnlock | sTeamMode | bTeamsEditable | idTeamInGroup | iTeamMaxMembers | bHasAttempts | sAccessOpenDate      | sDuration | sEndContestDate      | bShowUserInfos | sContestPhase | iLevel | bNoScore | groupCodeEnter |
+    | 50 | Course | null | 1                 | 0              | 0        | null    | 1                | 0              | 0                       | 1        | 0         | default     | 0               | 0           | 0             | 0           | All             | null           | null           | 100             | null      | 0              | null          | 0               | 0            | null                 | null      | null                 | 0              | Running       | null   | 0        | 0              |
+  And the table "items_strings" should stay unchanged
+  And the table "items_items" should be:
+    | idItemParent | idItemChild |
+    | 21           | 60          |
```

Comment:
If you do not specify the children, they should not be removed. not specified <> []  !!!

Reply:
fixed